src/fdb_kvdb.c

```diff
@@ -1559,10 +1559,24 @@ void fdb_kvdb_control(fdb_kvdb_t db, int cmd, void *arg)
         *(uint32_t *)arg = db->parent.sec_size;
         break;
     case FDB_KVDB_CTRL_SET_LOCK:
+#if __GNUC__
```

Comment:
Please replace all `__GNUC__` to this because when using ARMCC it has some new warning.
```suggestion
#if !defined(__ARMCC_VERSION) && defined(__GNUC__)
```